homeassistant/components/upcloud/__init__.py

```diff
@@ -155,6 +157,17 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     hass.data[DATA_UPCLOUD] = UpCloudHassData()
     hass.data[DATA_UPCLOUD].coordinators[entry.data[CONF_USERNAME]] = coordinator
 
+    # Set up service in device registry
+    _ = device_registry.async_get(hass).async_get_or_create(
```

Comment:
Seems like all this data is already available in the entity below, why not just move it into the device info there?

Reply:
Indirect device creation as kind of a side effect of an entity returning some device info doesn't seem to me as clean as explicitly creating the device and just returning identifiers or connections to "link" entities to it. But if the project's general opinion differs from mine or I've missed something, I can surely change that.

A downside of the implementation here is that the identifiers need to be duplicated.

Reply:
Removed the separate device registry setup, moved info to `device_info`.